### ISSUE TITLE
Request - Add readout of Pro Controller stick and trigger values

### ISSUE THREAD

**Issue by dotdissent** [TheHighTide/HTButtonTester#1]
What it says on the tin. Would like to easily test Pro Controller sticks and triggers and just being able to see the value readout of them would be perfect.

**Comment by TheHighTide** [comment]
I will work on it later today 😃

**Comment by TheHighTide** [comment]
Sorry for the wait but it looks like the current library I use for input `vpad.h` doesn't support the pro controller and only the gamepad. I am currently looking and asking around for a library I can switch to that supports it all. I have updated the app with a few changes but it might take a bit longer to fix this.

Again, sorry for the wait :D

**Comment by dotdissent** [comment]
No worries, I appreciate the effort!! :)

**Comment by dotdissent** [comment]
Suddenly recalled today that the triggers aren't analog 😂 So yeah, just looking for the X/Y values of the sticks. Or, even better, a visualizer if that sounds easy/fun to do. Lmk if you have a donation link and I'll send what I can your way 😊

**Comment by TheHighTide** [comment]
I will definitely work on it. Just wondering, does the app work with the pro controller at all or just gamepad?
Also here's a donation link: 😄 
https://ko-fi.com/thehightide

**Comment by TheHighTide** [comment]
I have found a library that supports the Wii controllers and WiiU controllers (not gamepad) and am starting to work on the update now. Expect the patch in 2 to 20 hours. :D

**Comment by TheHighTide** [comment]
Gosh I know this is probably frustrating but after going in a back and forth with another developer I learned that the library that I just spent 2ish hours implementing only supports **Wii** controllers and the **Wii** pro controller. I'm very sorry but until I can learn how to use another library it just won't be possible for me. I know what library I need to learn but its 12:15am right now and I'm tired lol. Sorry for the delay and I promise it will be done eventually. 😄

**Comment by dotdissent** [comment]
All good, I'm not worried about speed, sorry it's been a weird one to
figure out!

**Comment by TheHighTide** [comment]
I have got the pro controller working in the app now. Just one question before I start coding the features. You wanted a readout of the X/Y values of each joystick?

**Comment by dotdissent** [comment]
Correct! That way I can check for bad potentiometers in my sticks, the
values become erratic when they're physically degraded.

**Comment by TheHighTide** [comment]
I am now pushing the update! There is now advanced mode and basic mode. Advanced mode can be accessed by pressing ZL+ZR+A on the pro controller or gamepad. This will show X/Y values for both joysticks on both controllers. If you are in basic mode (default) it will only show text for the direction if the x/y value is greater than 0.5. Thanks for giving me this challenge and I will keep this issue open for 1 day as a sort of buffer period incase any more issues with this feature arise. If you have any questions about using the program or need more info, you can contact my discord with the username **higgytiggy**. 

Once again, thank you!

Note:
It can take up to 1-2 days for the update to appear on the homebrew store because they are slow😃

**Comment by dotdissent** [comment]
Thanks for this! Been super busy and finally got time to test it, found that the Pro Controller A and B buttons are reading swapped which also means you need to use ZL+ZR+B to enter Advanced mode.

**Comment by TheHighTide** [comment]
Oh shoot lemme fix that real quick I noticed that issue and thought I got all of it but I guess not. It will be an easy fix though. Give me a few hours and I will send out a patch :D
